src/mbgl/mtl/context.cpp

```diff
@@ -123,11 +123,18 @@ UniqueShaderProgram Context::createProgram(std::string name,
         std::move(name), backend, std::move(vertexFunction), std::move(fragmentFunction));
 }
 
+MTLTexturePtr Context::createMetalTexture(MTLTextureDescriptorPtr textureDescriptor) const {
+    return NS::TransferPtr(backend.getDevice()->newTexture(textureDescriptor.get()));
```

Comment:
We should be collecting metrics in the metal context like the GL context does, it'll be helpful for developing new Metal render tests and doing general profiling.

Reply:
That's a good idea.  Would you open an Issue and let's figure out what metrics we want.
What are we already collecting, like frame rate?

Reply:
The GL context tracks the number of allocated textures and the size of requested texture memory. [Here](https://github.com/maplibre/maplibre-native/blob/b792309d6adc0ec1d3aad6f292f1e08df96061e2/include/mbgl/gfx/rendering_stats.hpp#L10-L18) is everything the renderer tracks.